File: tests/fixtures/_highlight-corpus/sample.swift
// Swift sample
import Foundation

final class Counter {
    private(set) var value: Int = 0
    func increment(_ n: Int = 1) -> Int {
        value += n
        return value
    }
}

@main
struct App {
    static func main() throws {
        let url = URL(fileURLWithPath: "input.txt")
        let text = try String(contentsOf: url, encoding: .utf8)
        let counter = Counter()
        for line in text.split(separator: "\n") {
            if !line.isEmpty {
                _ = counter.increment(line.count)
            }
        }
        print("total bytes: \(counter.value)")
    }
}

File: tests/fixtures/_highlight-corpus/sample.py
# Python sample
"""A small illustrative module."""

from __future__ import annotations

import asyncio
from dataclasses import dataclass
from pathlib import Path


@dataclass(frozen=True)
class Counter:
    value: int = 0

    def incremented(self, n: int = 1) -> "Counter":
        return Counter(self.value + n)


async def main(path: Path) -> int:
    text = path.read_text(encoding="utf-8")
    counter = Counter()
    for line in text.splitlines():
        if line:
            counter = counter.incremented(len(line))
    print(f"total bytes: {counter.value}")
    return 0


if __name__ == "__main__":
    raise SystemExit(asyncio.run(main(Path("input.txt"))))
